code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -0,0 +1,563 @@
+					extra_rulesets_amount++
+
+	if (drafted_rules.len > 0 && picking_roundstart_rule(drafted_rules))
+		if (extra_rulesets_amount > 0)//we've got enough population and threat for a second rulestart rule
+			for (var/datum/dynamic_ruleset/roundstart/rule in drafted_rules)
+				if (rule.cost > threat)
+					drafted_rules -= rule
+			if (drafted_rules.len > 0 && picking_roundstart_rule(drafted_rules))
+				if (extra_rulesets_amount > 1)//we've got enough population and threat for a third rulestart rule
+					for (var/datum/dynamic_ruleset/roundstart/rule in drafted_rules)
+						if (rule.cost > threat)
+							drafted_rules -= rule
+	else
+		return FALSE
+	return TRUE
+
+/datum/game_mode/dynamic/proc/picking_roundstart_rule(var/list/drafted_rules = list())
+	var/datum/dynamic_ruleset/roundstart/starting_rule = pickweight(drafted_rules)
+
+	if (starting_rule)
+		log_admin("Picking a [istype(starting_rule, /datum/dynamic_ruleset/roundstart/delayed/) ? " delayed " : ""] ruleset...<font size='3'>[starting_rule.name]</font>!")
+
+		roundstart_rules -= starting_rule
+		drafted_rules -= starting_rule
+
+		if (istype(starting_rule, /datum/dynamic_ruleset/roundstart/delayed/))
+			spend_threat(starting_rule.cost)
+			return pick_delay(starting_rule)
+
+		spend_threat(starting_rule.cost)
+		threat_log += "[worldtime2text()]: Roundstart [starting_rule.name] spent [starting_rule.cost]"
+		if (starting_rule.pre_execute())//this should never fail since ready() returned 1
+			executed_rules += starting_rule
+			if (starting_rule.persistent)
+				current_rules += starting_rule
+			for(var/mob/M in starting_rule.assigned)
+				candidates -= M
+				for (var/datum/dynamic_ruleset/roundstart/rule in roundstart_rules)
+					rule.candidates -= M//removing the assigned players from the candidates for the other rules
+					if (!rule.ready())
+						drafted_rules -= rule//and removing rules that are no longer elligible
+			return TRUE
+	return FALSE
+
+/datum/game_mode/dynamic/proc/pick_delay(var/datum/dynamic_ruleset/roundstart/delayed/rule)
+	rule.candidates = GLOB.player_list.Copy()
+	rule.trim_candidates()
+	if(!rule.pre_execute())
+		return FALSE
+	addtimer(CALLBACK(src, .proc/execute_delayed, rule), rule.delay)
+	return TRUE
+
+/datum/game_mode/dynamic/proc/execute_delayed(var/datum/dynamic_ruleset/roundstart/delayed/rule)
+	if(rule.execute())
+		executed_rules += rule
+		if (rule.persistent)
+			current_rules += rule
+		return TRUE
+	
+/datum/game_mode/dynamic/proc/picking_latejoin_rule(var/list/drafted_rules = list())
+	var/datum/dynamic_ruleset/latejoin/latejoin_rule = pickweight(drafted_rules)
+	if (latejoin_rule)
+		if (!latejoin_rule.repeatable)
+			latejoin_rules = remove_rule(latejoin_rules,latejoin_rule.type)
+		spend_threat(latejoin_rule.cost)
+		threat_log += "[worldtime2text()]: Latejoin [latejoin_rule.name] spent [latejoin_rule.cost]"
+		if (latejoin_rule.execute())//this should never fail since ready() returned 1
```

Comment:
add an else + crash or stack_trace call indicating what rules failed, because a comment, suprisingly, often causes failures

Reply:
I did else + stack_trace _and comments do stop failures if you comment out the code that fails._